proto/frequenz/api/common/v1/microgrid/communication_components.proto

```diff
@@ -0,0 +1,71 @@
+// Frequenz microgrid communication components definitions.
+//
+// Copyright:
+// Copyright 2023 Frequenz Energy-as-a-Service GmbH
+//
+// License:
+// MIT
+
+syntax = "proto3";
+
+package frequenz.api.common.v1.microgrid.communication_components;
+
+
+// CommunicationComponent represents a communication component within a
+// microgrid.
+//
+// Communication components are networked devices responsible for enabling
+// communication between electrical components, sensors, controllers, and other
+// networked equipment.
+message CommunicationComponent {
+  // Unique identifier of the communication component.
+  uint64 id = 1;
+
+  // Unique identifier of the parent microgrid.
+  uint64 microgrid_id = 2;
+
+  // Human-readable name of the communication component.
+  string name = 3;
+
+  // Category identifying the type of the communication component.
+  CommunicationComponentCategory category = 4;
+
+  // List of IP addresses assigned to this communication component.
+  repeated string ip_addresses = 5;
+}
+
+// CommunicationComponentCategory enumerates possible types of communication
```

Comment:
Nit: I would recommend putting the name of the struct in backticks, so that it gets formatted accordingly in the auto-generated docs.